src/geodesic_on_the_sphere.jl

```diff
@@ -0,0 +1,258 @@
+        target::Vector{<:MP.AbstractPolynomial}) where {T<:Number}
+        GeodesicOnTheSphere{T}(
+            convert(Vector{FP.Polynomial{T}}, start),
+            convert(Vector{FP.Polynomial{T}}, target))
+    end
+end
+
+#
+# CONSTRUCTORS
+#
+function GeodesicOnTheSphere(
+    start::Vector{FP.Polynomial{T}},
+    target::Vector{FP.Polynomial{S}}) where {T<:Number, S<:Number}
+    GeodesicOnTheSphere{promote_type(S,T)}(start,target)
+end
+function GeodesicOnTheSphere(
+    start::Vector{<:MP.AbstractPolynomial{T}},
+    target::Vector{<:MP.AbstractPolynomial{S}}) where {T<:Number, S<:Number}
+    P = promote_type(S, T)
+    GeodesicOnTheSphere{T}(
+        convert(Vector{FP.Polynomial{T}}, start),
+        convert(Vector{FP.Polynomial{T}}, target))
+end
+function GeodesicOnTheSphere(
+    start::MP.AbstractPolynomial{T},
+    target::MP.AbstractPolynomial{S}) where {T,S}
+    U = promote_type(S, T)
+    s, t = convert(Vector{FP.Polynomial{T}}, [start, target])
+    GeodesicOnTheSphere{T}([s], [t])
+end
+
+#
+# SHOW
+#
+function Base.show(io::IO, H::GeodesicOnTheSphere)
+    start = join(string.(H.start), ", ")
+    target = join(string.(H.target), ", ")
+    println(io, typeof(H),
+    " The homotopy is given by the spherical geodesic from start/|start| (t=1) to target/|target| (t=0). ",
+    "Here: target = [", target, "], start = [", start,"].")
+end
+
+
+#
+# EQUALITY
+#
+function ==(H1::GeodesicOnTheSphere, H2::GeodesicOnTheSphere)
+    H1.start == H2.start && H1.target == H2.target
+end
+function Base.isequal(H1::GeodesicOnTheSphere, H2::GeodesicOnTheSphere)
+    Base.isequal(H1.start, H2.start) && Base.isequal(H1.target, H2.target)
+end
+
+#
+# PROMOTION AND CONVERSION
+#
+function Base.promote_rule(
+    ::Type{GeodesicOnTheSphere{T}},
+    ::Type{GeodesicOnTheSphere{S}}) where {S<:Number,T<:Number}
+    GeodesicOnTheSphere{promote_type(T,S)}
+end
+
+function Base.convert(
+    ::Type{GeodesicOnTheSphere{T}},
+    H::GeodesicOnTheSphere) where {T}
+    GeodesicOnTheSphere{T}(H.start, H.target)
+end
+
+#
+# EVALUATION + DIFFERENTATION
+#
+function evaluate!(u::AbstractVector{T}, H::GeodesicOnTheSphere{T}, x::Vector{T}, t::Number) where {T<:Number}
+    inverse_norm_start = one(T)/FP.weylnorm(H.start)
+    inverse_norm_target = one(T)/FP.weylnorm(H.target)
+    α=acos(FP.weyldot(H.start,H.target)  * inverse_norm_start * inverse_norm_target)
+    λ_1 = sin(t*α)/sin(α)
+    λ_2 = cos(t*α) - λ_1*cos(α)
+
+    map!(u, H.target, H.start) do f, g
+        inverse_norm_start * λ_1 * FP.evaluate(g, x) + inverse_norm_target * λ_2 * FP.evaluate(f, x)
+    end
+end
+
+function evaluate(H::AbstractPolynomialHomotopy{T}, x::Vector{T}, t::Number) where {T<:Number}
+    evaluate!(zeros(H.target, T), H, x,  t)
+end
+(H::GeodesicOnTheSphere)(x,t) = evaluate(H,x,t)
+
+
+function weylnorm(H::GeodesicOnTheSphere{T})  where {T<:Number}
+    function (t)
+         1
+     end
+end
+
+
+function differentiate(F::Vector{FP.Polynomial{T}}) where {T<:Number}
```

Comment:
This overwrites the same function in StraightLine, so it would be enough to remove it